src/TestProblem/ELBDM/GaussianWavePacket/Init_TestProb_ELBDM_GaussianWavePacket.cpp

```diff
@@ -185,9 +189,45 @@ void SetGridIC( real fluid[], const double x, const double y, const double z, co
    const double Gau_Theta1 = -0.5*acos(  pow( Gau_Const1, -0.5 )  );
    const double Gau_Theta2 = 0.5*pow( dr1, 2.0 )*ELBDM_ETA*Time/(  pow( ELBDM_ETA*SQR(Gau_Width), 2.0) + SQR(Time)  )
                              + Gau_v0*ELBDM_ETA*dr2;
-
-   fluid[REAL] = Gau_Const2*cos( Gau_Theta1 + Gau_Theta2 );
-   fluid[IMAG] = Gau_Const2*sin( Gau_Theta1 + Gau_Theta2 );
+   double Re, Im;
+
+   Re = Gau_Const2*cos( Gau_Theta1 + Gau_Theta2 );
+   Im = Gau_Const2*sin( Gau_Theta1 + Gau_Theta2 );
+
+   // Approximate periodic BC by including Gaussian wave packets outside the box
+   if ( Gau_PeriodicN > 0 ){
+      double Gau_Center_p, dr1_p, dr2_p, Gau_Const2_p, Gau_Theta2_p;
+      double Gau_Center_m, dr1_m, dr2_m, Gau_Const2_m, Gau_Theta2_m;
+
+      for (int n=1; n<Gau_PeriodicN+1; n++){
+         // Gaussian wave packet image for periodic BC on plus(+) direction
+         Gau_Center_p      = Gau_Center + n*amr->BoxSize[Gau_XYZ];
+         dr1_p             = r -     Gau_v0*Time - Gau_Center_p;
+         dr2_p             = r - 0.5*Gau_v0*Time - Gau_Center_p;
+         Gau_Const2_p      = pow( SQR(Gau_Width)*M_PI*Gau_Const1, -0.25 )
+                             *exp(  -0.5*pow( dr1_p/Gau_Width, 2.0 )/Gau_Const1  );
+         Gau_Theta2_p      = 0.5*pow( dr1_p, 2.0 )*ELBDM_ETA*Time/(  pow( ELBDM_ETA*SQR(Gau_Width), 2.0) + SQR(Time)  )
+                             + Gau_v0*ELBDM_ETA*dr2_p;
+
+         Re += Gau_Const2_p*cos( Gau_Theta1 + Gau_Theta2_p );
+         Im += Gau_Const2_p*sin( Gau_Theta1 + Gau_Theta2_p );
+
+         // Gaussian wave packet image for periodic BC on minus(-) direction
+         Gau_Center_m      = Gau_Center - n*amr->BoxSize[Gau_XYZ];
+         dr1_m             = r -     Gau_v0*Time - Gau_Center_m;
+         dr2_m             = r - 0.5*Gau_v0*Time - Gau_Center_m;
+         Gau_Const2_m      = pow( SQR(Gau_Width)*M_PI*Gau_Const1, -0.25 )
+                             *exp(  -0.5*pow( dr1_m/Gau_Width, 2.0 )/Gau_Const1  );
+         Gau_Theta2_m      = 0.5*pow( dr1_m, 2.0 )*ELBDM_ETA*Time/(  pow( ELBDM_ETA*SQR(Gau_Width), 2.0) + SQR(Time)  )
+                             + Gau_v0*ELBDM_ETA*dr2_m;
+
+         Re += Gau_Const2_m*cos( Gau_Theta1 + Gau_Theta2_m );
+         Im += Gau_Const2_m*sin( Gau_Theta1 + Gau_Theta2_m );
+      }
+   } // if ( Gau_PeriodicN > 0 )
```

Comment:
To minimize redundant codes, how about combining the original and image wave packets as follows?
```
   const double Gau_Const1 = 1.0 + pow(  Time / ( ELBDM_ETA*SQR(Gau_Width) ), 2.0  );
   const double Gau_Theta1 = -0.5*acos(  pow( Gau_Const1, -0.5 )  );
   double Re=0.0, Im=0.0;

// n=0, m=0: original wave packet
// n>0, m=0/1: images for periodic BC on the plus(+)/minus(-) direction
   for (int n=0; n<Gau_PeriodicN+1; n++) {
   for (int m=0; m<((n==0)?1:2); m++) {
      const double Center     = Gau_Center + n*(1-2*m)*amr->BoxSize[Gau_XYZ];
      const double dr1        = r -     Gau_v0*Time - Center;
      const double dr2        = r - 0.5*Gau_v0*Time - Center;
      const double Gau_Const2 = pow( SQR(Gau_Width)*M_PI*Gau_Const1, -0.25 )
                                *exp(  -0.5*pow( dr1/Gau_Width, 2.0 )/Gau_Const1  );
      const double Gau_Theta2 = 0.5*pow( dr1, 2.0 )*ELBDM_ETA*Time/(  pow( ELBDM_ETA*SQR(Gau_Width), 2.0) + SQR(Time)  )
                                + Gau_v0*ELBDM_ETA*dr2;

      Re += Gau_Const2*cos( Gau_Theta1 + Gau_Theta2 );
      Im += Gau_Const2*sin( Gau_Theta1 + Gau_Theta2 );
   }}
```

Reply:
Done.